### PR TITLE
Convert from debug_mode to verbosity

### LINKED ISSUE #68: Fully implement new VERBOSITY settings
### Desired capability or behavior.

With the introduction of the `Settings` category to the variable hierarchy, we now have a official way to control output via the `VERBOSITY` flag. This variable now needs to be propagated through the code. A few immediate fixes:
- Replace the `debug_mode` flag everywhere it appears, and replace with the appropriate `VERBOSITY` level. This includes modifying fortran_to_aviary to no longer hardcode adding in `debug_mode`, and all of the special handling needed to deal with this add-on variable
- Wrap all print statements in the code under checks for a minimum threshold `VERBOSITY` level

`VERBOSITY` enum values are ints, so greater than/less than comparators should be used for logic statements whenever possible (e.g. only print specific info if `VERBOSITY` is greater than 2, etc.)

### Is your feature request related to a problem? Please describe.

_No response_

### Associated Bug Report

_No response_

### PR DESCRIPTION
**Pull request by crecine**
### Summary

converting from ad hoc debug mode implementations with a unified verbosity

### Related Issues

- Resolves #68 

### Backwards incompatibilities

None

### New Dependencies

None

### CODE REVIEW

**Comment by crecine** [comment]
I initially converted everything to Verbosity.DEBUG or Verbosity.QUIET, but would appreciate feedback on what should be BRIEF or VERBOSE

File: aviary/utils/Fortran_to_Aviary.py
Comment (Kenneth-T-Moore) [inline aviary/utils/Fortran_to_Aviary.py:41]: I feel like the default ought to be Verbosity.Brief.  There is a certain amount of information that we probably should print out for the user (I think Overrides fall into this category.)  Then, if they want to clear everything out, they can go quiet (and we might even suppress openmado warnings for that one.)

I could also be persuaded that Verbose should be default. I think granularizing our messages between Brief and Verbose will take some thought.  Of course, that leaves Debug for the kitchen sink.

Comment (crecine) [reply aviary/utils/Fortran_to_Aviary.py:41]: hmm, maybe:
debug - everything
verbose - default
brief - only warnings
quiet - nothing

Comment (crecine) [reply aviary/utils/Fortran_to_Aviary.py:41]: or:
debug - everything
verbose - optimizer output
brief - default
quite - nothing

File: aviary/mission/gasp_based/ode/time_integration_base_classes.py
Comment (jkirk5) [inline aviary/mission/gasp_based/ode/time_integration_base_classes.py:190]: The "or True" should get removed

File: aviary/utils/Fortran_to_Aviary.py
Comment (jkirk5) [reply aviary/utils/Fortran_to_Aviary.py:41]: I lean towards that second description. However I think we need to let the user know if the optimization was successful or not in that case, even if we don't show them the whole optimizer output

Comment (jkirk5) [reply aviary/utils/Fortran_to_Aviary.py:41]: I agree with @Kenneth-T-Moore about quiet operation suppressing non-critical warnings. That was kind of the intent for that mode, to suppress literally everything that wasn't a show-stopping bug. That way big parametric studies won't generate pages of terminal output, which really isn't helpful to anyone.

File: aviary/utils/Fortran_to_Aviary.py
Comment (jkirk5) [inline aviary/utils/Fortran_to_Aviary.py:615]: I like the simplified "on-off" verbose switch for the command args here

File: aviary/utils/process_input_decks.py
Comment (jkirk5) [inline aviary/utils/process_input_decks.py:141]: I think it is really important for users to know if their inputs are getting skipped. But, we don't need to share the unconverted FORTRAN variables either... maybe we keep it debug for now. It would be neat if we could filter out all FLOPS and GASP variables.

File: aviary/mission/gasp_based/phases/time_integration_traj.py
Comment (jkirk5) [inline aviary/mission/gasp_based/phases/time_integration_traj.py:236]: "self", just in case we need this code again

File: aviary/mission/gasp_based/test/test_idle_descent_estimation.py
Comment (jkirk5) [inline aviary/mission/gasp_based/test/test_idle_descent_estimation.py:21]: Maybe we should make an issue to add either QUIET or BRIEF to all tests to reduce unnecessary outputs. How does everyone feel about test outputs?

Comment (crecine) [reply aviary/mission/gasp_based/test/test_idle_descent_estimation.py:21]: I'm not sure if it is possible, but it would be nice if running testflo for the entire repository was QUIET and then failed tests were rerun with verbose.

File: aviary/utils/process_input_decks.py
Comment (crecine) [reply aviary/utils/process_input_decks.py:141]: changing to not QUIET for now, we can look into more detailed filtering as a follow on.

**Review by Kenneth-T-Moore — APPROVED** [review]

File: aviary/interface/methods_for_level2.py
Comment (johnjasa) [inline aviary/interface/methods_for_level2.py:1590]: Because `verbosity` is only ever an enum value, not a list, this logic doesn't hold. There wouldn't be a way to have the user set a custom amountn of output. Here the default verbosity results in `debug_print` being turned on; I'd suggest  it only being on for `VERBOSE` and `DEBUG` if you keep the default as `BRIEF`. Also you can just remove the list logic.

**Review by johnjasa — APPROVED** [review]
Minor note and change request; I don't want it to hold up approval. Thanks for this!

**Pull request merged**
